src/message.ts

```diff
@@ -21,12 +21,23 @@ export class Message {
      * Number representing the message version.
      */
     public version: number;
+    /**
+     * The library or tool that was used to sign the message.
+     */
+    public signer: string;
 
-    constructor(options: { data: Uint8Array; signature?: Uint8Array; address?: IAddress; version?: number }) {
+    constructor(options: {
+        data: Uint8Array;
+        signature?: Uint8Array;
+        address?: IAddress;
+        version?: number;
+        signer?: string;
+    }) {
         this.data = options.data;
         this.signature = options.signature;
         this.address = options.address;
         this.version = options.version || DEFAULT_MESSAGE_VERSION;
+        this.signer = options.signer ?? SDK_JS_SIGNER;
```

Comment:
On the constructor, here, it's OK to put the default is missing. But on unpack, we should use a different placeholder, I think. E.g. `unknown`.

Reply:
set to `unknown` when unpacking a message with no `signer` field

Reply:
indeed, here the `||` operator would have worked just fine, will replace it to stay consistent.  Above I am using the `||` operator because someone could have passed the value `0` which would be a valid number, but it's not a valid message version.

Bellow, this line `signer: message.signer ? message.signer : SDK_JS_SIGNER` could simply be `signer: message.signer` since `packMessage` needs an object of type `Message` and the signer always get's initialized in the constructor. The same applies for the `version`